### ISSUE TITLE
URL of token is very large for mobiles

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roundcubevnz/myroundcube#332]
```
I did a token for read/write. In my environment is:
https://10.70.1.48:443/webmail/?_task=dummy&_action=plugin.calendar_showlayer&_u
serid=26&_ct=2569fb7815077a54982116b2485653f6

This string have 125 characteres. In the calendar of Android, it supports only 
100 characteres. So, is possible reduce the lenght of URL?

Best,
Claudio
```

Original issue reported on code.google.com by `filh...@gmail.com` on 2 May 2012 at 7:08

**Comment by GoogleCodeExporter** [comment]
```
Can't you just use mod_rewrite (Apache)?
```

Original comment by `rol...@liebl.ath.cx` on 2 May 2012 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
http://www.websiteoptimization.com/speed/tweak/abbreviate/
```

Original comment by `rol...@liebl.ath.cx` on 2 May 2012 at 8:26

**Comment by GoogleCodeExporter** [comment]
```
Mods for .htaccess in Roundcube root
{{{

<IfModule mod_rewrite.c>
RewriteEngine On
RewriteRule ^favicon.ico$ skins/default/images/favicon.ico
# security rules
RewriteRule .svn/ - [F]
RewriteRule ^README|INSTALL|LICENSE|SQL|bin|CHANGELOG$ - [F]
# calendar feeds / ics
RewriteRule ^feeds/([0-9]+)/([a-z0-9]+)$ 
index.php?_task=dummy&_action=plugin.calendar_showlayer&_userid=$1&_ct=$2
RewriteRule ^ics/([0-9]+)/([a-z0-9]+)$ 
index.php?_task=dummy&_action=plugin.calendar_showlayer&_userid=$1&_ct=$2&_ics=1
</IfModule>

}}}

https://10.70.1.48:443/webmail/?_task=dummy&_action=plugin.calendar_showlayer&_u
serid=26&_ct=2569fb7815077a54982116b2
485653f6

... becomes ...

https://10.70.1.48:443/webmail/feeds/26/2569fb7815077a54982116b2485653f6 (for 
RSS feeds)

... and ...

https://10.70.1.48:443/webmail/ics/26/2569fb7815077a54982116b2485653f6 (for ICS 
feeds)

I'll add a config option to calendar config to show abbreviated URLs in 
calendar config if configured.


```

Original comment by `rol...@liebl.ath.cx` on 3 May 2012 at 5:56
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
My hero! \o/
I tried many options, but without success. I broke the PHP interpretation.
Thanks, Rosali
```

Original comment by `filh...@gmail.com` on 3 May 2012 at 9:37

**Comment by GoogleCodeExporter** [comment]
```
Fixed in calendar v8.8.

Note new calendar config value 'cal_short_urls'.

.htaccess adjustments:

{{{

<IfModule mod_rewrite.c>
RewriteEngine On
RewriteRule ^favicon.ico$ skins/default/images/favicon.ico
# security rules
RewriteRule .svn/ - [F]
RewriteRule ^README|INSTALL|LICENSE|SQL|bin|CHANGELOG$ - [F]
# calendar
RewriteRule ^rc/([0-9]+)/([a-z0-9]+)$ 
index.php?_task=dummy&_action=plugin.calendar_showlayer&_userid=$1&_ct=$2
RewriteRule ^ics/([0-9]+)/([a-z0-9]+)$ 
index.php?_task=dummy&_action=plugin.calendar_showlayer&_userid=$1&_ct=$2&_ics=1
</IfModule>

}}}
```

Original comment by `rol...@liebl.ath.cx` on 10 May 2012 at 7:53
- Changed state: **Fixed**
